header/actor-zeta/detail/rtt.hpp

```diff
@@ -0,0 +1,275 @@
+#pragma once
+
+#include <actor-zeta/detail/rtt_management.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace actor_zeta { namespace detail {
+
+    inline void dummy(std::initializer_list<int>) {}
+
+#define EXPAND_VARIADIC(expression) \
+    dummy({(expression, 0)...})
+
+    template<class...>
+    struct padded_size;
+
+    template<>
+    struct padded_size<> {
+        static constexpr std::size_t value = 0;
+    };
+
+    template<class Head>
+    struct padded_size<Head> {
+        static constexpr std::size_t value = sizeof(Head);
+    };
+
+    template<class Head, class Mid, class... Tail>
+    struct padded_size<Head, Mid, Tail...> {
+        static constexpr std::size_t value = (sizeof(Head) + alignof(Mid) - 1) / alignof(Mid) * alignof(Mid) + padded_size<Mid, Tail...>::value;
+    };
+
+    // run-time tuple
+    class rtt final {
+    private:
+        using object_info_container_type = std::vector<management::object_info_t>;
+
+    public:
+        using size_type = typename object_info_container_type::size_type;
+
+    public:
+        template<typename Head, typename... Tail,
+                 typename =
+                     type_traits::enable_if_t<
```

Comment:
Is it necessary to use `enable_if` here? If I understand your intent correctly, you want to avoid conflict between this constructor and copy/move ctor. But ordinary function (ctor) should have priority over template. The only difference I can think of is `volatile` overload.